libfprint/drivers/goodixtls/5395/goodix_device.c

```diff
@@ -258,3 +268,94 @@ gboolean fpi_goodix_device_reset(FpDevice *dev, guint8 reset_type, gboolean irq_
     GError *error = NULL;
     return fpi_goodix_device_send(dev, message, TRUE, 500, FALSE, &error);
 }
+
+gboolean fpi_goodix_device_gtls_connection(FpDevice *dev, GError *error) {
+    fp_dbg("fpi_goodix_device_gtls_connection()");
+    fp_dbg("Starting GTLS handshake");
+    FpiGoodixDevice *self = FPI_GOODIX_DEVICE(dev);
+    FpiGoodixDevicePrivate *priv = fpi_goodix_device_get_instance_private(self);
+    priv->gtls_params = fpi_goodix_device_gtls_init_params();
+
+    // Hello phase
+    fpi_goodix_gtls_create_hello_message(priv->gtls_params);
+    fp_dbg("client_random: %s", fpi_goodix_protocol_data_to_str(priv->gtls_params->client_random->data, priv->gtls_params->client_random->len));
+    fp_dbg("client_random_len: %02x", priv->gtls_params->client_random->len);
+    fpi_goodix_device_send_mcu(dev, 0xFF01, priv->gtls_params->client_random);
+    priv->gtls_params->state = 2;
```

Comment:
'2' value is not clear to know what the state. I would like create the enum for describe gtls state.

Reply:
I don't what this state means. I just copied them from the python.
Do you know what they mean?

Reply:
ohh, got it. I think this variable store current step of handshake. We can use our own constants: 
INITILIZE_HANDSHAKE
SEND_HELLO
IDENTIFY
COMPLETE_HANDSHAKE

Also, we can use SSM for handle this.

I'm going to  merge PR with this implementation, but want to/can rewrite it with using SSM.